sbt-plugin/src/main/scala/ch/epfl/scala/AnalyzeDependencyGraph.scala

```diff
@@ -0,0 +1,220 @@
+package ch.epfl.scala
+
+import java.nio.file.Paths
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.sys.process._
+import scala.util.Properties
+import scala.util.Try
+
+import ch.epfl.scala.GithubDependencyGraphPlugin.autoImport._
+import ch.epfl.scala.githubapi._
+import gigahorse.FullResponse
+import gigahorse.HttpClient
+import gigahorse.support.asynchttpclient.Gigahorse
+import sbt._
+import sbt.internal.util.complete._
+import sjsonnew.shaded.scalajson.ast.unsafe.JArray
+import sjsonnew.shaded.scalajson.ast.unsafe.JField
+import sjsonnew.shaded.scalajson.ast.unsafe.JObject
+import sjsonnew.shaded.scalajson.ast.unsafe.JString
+import sjsonnew.support.scalajson.unsafe.{Parser => JsonParser}
+
+object AnalyzeDependencyGraph {
+
+  object Model {
+    val help =
+      "download and display CVEs alerts from Github, and analyze them against dependencies (use hub or gh local config or GIT_TOKEN env var to authenticate, requires githubGenerateSnapshot)"
+
+    def blue(str: String): String = s"\u001b[34m${str}\u001b[0m"
```

Comment:
For me, it is easier to read the main implementation first and then the details. First, to get a broad view of what the command does, and then to see how each part is implemented.

Something like this:
```scala
// First the command definition
val commands = Seq(
  Command(
    AnalyzeDependencies,
    (AnalyzeDependenciesUsage, AnalyzeDependenciesDetail),
    AnalyzeDependenciesDetail
  )(extractPattern)(analyzeDependencies)

// then the command parser and implementation
private def extractPattern(state: State): Parser[AnalysisParams] = ???

private def analyzeDependencies(state: State, params: AnalysisParams): State = ???

// then the details of their implementations
private def downloadAlerts(state: State, repo: String): Try[State] = ???

private def analyzeCves(state: State): State = ???

// at the very bottom, the small helper functions
private def getStateOrWarn[T](state: State, key: AttributeKey[T], what: String, command: String): Option[T] = ???

private def getGithubToken(file: File): Option[String] = ???

def blue(str: String): String = ???
```